sale_promotion_rule/models/sale_promotion_rule.py

```diff
@@ -95,6 +105,37 @@ class SalePromotionRule(models.Model):
     _sql_constraints = [
         ('code_unique', 'UNIQUE (code)', _('Discount code must be unique !'))]
 
+    @api.constrains("discount_product_id", "promo_type", "discount_type")
+    def _validate_promotion_product_id(self):
```

Comment:
Please follow https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#models for method name.

Same here under